src/workerd/io/trace-common.h

```diff
@@ -276,5 +577,19 @@ struct Span {
         endTime(startTime) {}
 };
 
+// ======================================================================================
+// The base class for both the original legacy Trace (defined in trace-legacy.h)
+// and the new StreamingTrace (defined in trace-streaming.h)
+class TraceBase {
```

Comment:
Nit: (and we can remove all virtual declarations...)

```
class TraceBase virtual
```

Reply:
Compiler error using that